sections/home-hero-section/index.js

```diff
@@ -0,0 +1,139 @@
+import { rem } from 'polished'
+import Typist from 'react-typist'
+import styled from 'styled-components'
+
+import { ApplyLink, Heading, Text } from 'components'
+import 'react-typist/dist/Typist.css'
+import { TABLET_LARGE_SIZE, MOBILE_SIZE } from 'styles/constants'
+import styles from 'styles/Home.module.css'
+
+const StyledSection = styled.section`
+  display: flex;
+  flex-direction: column;
+  max-width: 1200px;
+  margin: 0 auto;
+  padding: 0 144px;
+
+  @media only screen and (max-width: ${TABLET_LARGE_SIZE}) {
+    height: 100vh;
+    justify-content: center;
+    padding: 0 80px;
+  }
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
+    margin: 142px 16px 0;
+    padding: 0 20px;
+  }
+`
+
+const HeroContent = styled.div`
+  height: ;
+`
+
+const StyledHeading = styled(Heading)`
+  font-size: ${rem('48px')};
+  line-height: 1.25;
+  margin-top: 7em;
+  margin-bottom: 1.1em;
+
+  @media only screen and (max-width: ${TABLET_LARGE_SIZE}) {
+    margin-top: 3.125em;
+  }
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
+    text-align: left;
+    margin-top: 3em;
+    margin-bottom: 0.75em;
+  }
+`
+
+const StyledText = styled(Text)`
+  color: #ffffff;
+  font-family: Montserrat;
+  font-size: ${rem('26px')};
+  line-height: 1.5;
+  text-align: left;
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
+    font-size: ${rem('24px')};
+  }
+`
+
+const HeroText = styled(StyledText)`
+  font-weight: 400;
+  margin: 2em 0;
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
+    margin-top: 1.5em;
+    margin-bottom: 2.7em;
+  }
+`
+
+const HeroTextExtra = styled(StyledText)`
+  display: block;
+  font-weight: 600;
+  margin-top: 0;
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
+    margin-top: 2em;
+  }
+`
+
+const HeroApplyLink = styled(ApplyLink)`
+  max-width: 380px;
+  margin-bottom: 6.5em;
+
+  @media only screen and (max-width: ${TABLET_LARGE_SIZE}) {
+    margin-bottom: 8.5em;
+  }
+
+  @media only screen and (max-width: ${MOBILE_SIZE}) {
```

Comment:
General tips for writing responsive CSS:

Start with the simplest, most generic styles possible (usually mobile), then add media queries and/or JS for screens and devices that allow for more complex scenarios. In terms of media queries, this means using `min-width` instead of `max-width`.

Reply:
I've refactored to defined styles starting with mobile and going up as opposed to the other way around. Also defined some constants for some media queries to make reading the code a bit easier.